dev/WebView2/WebView2.cpp

```
@@ -0,0 +1,1942 @@
+        [this](auto const&, winrt::CoreWebView2WebMessageReceivedEventArgs const& args)
+        {
+            // Fire the MUXC side NavigationCompleted event when the CoreWebView2 event is received.
+            FireWebMessageReceived(args);
+        }});
+
+    m_coreMoveFocusRequestedRevoker = m_coreWebViewController.MoveFocusRequested(winrt::auto_revoke, {
+        [this](auto const&, const winrt::CoreWebView2MoveFocusRequestedEventArgs& args)
+        {
+            winrt::CoreWebView2MoveFocusReason moveFocusRequestedReason{ args.Reason() };
+
+            if (moveFocusRequestedReason == winrt::CoreWebView2MoveFocusReason::Next ||
+                moveFocusRequestedReason == winrt::CoreWebView2MoveFocusReason::Previous)
+            {
+                winrt::FocusNavigationDirection xamlDirection{ moveFocusRequestedReason == winrt::CoreWebView2MoveFocusReason::Next ?
+                                                                   winrt::FocusNavigationDirection::Next : winrt::FocusNavigationDirection::Previous };
+                winrt::FindNextElementOptions findNextElementOptions;
+
+                auto contentRoot = [this]()
+                {
+                    if (auto thisXamlRoot = try_as<winrt::IUIElement10>())
+                    {
+                        if (auto xamlRoot = thisXamlRoot.XamlRoot())
+                        {
+                            return xamlRoot.Content();
+                        }
+                    }
+                    return winrt::Window::Current().Content();
+                }();
+
+                if (contentRoot)
+                {
+                    findNextElementOptions.SearchRoot(contentRoot);
+                    winrt::DependencyObject nextElement = [=]() {
+                        if (SharedHelpers::Is19H1OrHigher())
+                        {
+                            return winrt::FocusManager::FindNextElement(xamlDirection, findNextElementOptions);
```

>return winrt::FocusManager::FindNextElement(xamlDirection, findNextElementOptions); [](http://example.com/codeflow?start=28&length=83)

This may fail for WV2 in OS Island scenarios (when supported) due to:
Task 31103141: FocusManager APIs that don't take FocusNavigationOptions fail silently on Desktop Xaml apps

---

same below (line 603)

---

That's why this code is conditional. I ran into that and changing to the no-parameter version downlevel was needed to address it.